habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -200,13 +200,7 @@ def get_observation(
     ) -> VisualObservation:
         obs = cast(Optional[VisualObservation], sim_obs.get(self.uuid, None))
         check_sim_obs(obs, self)
-        # make semantic observation a 3D array
-        if isinstance(obs, np.ndarray):
-            obs = np.expand_dims(
-                obs, axis=2
-            ).astype(np.int32)
-        else:
-            obs = obs.unsqueeze(-1)
+        obs = obs[..., None].astype(np.int32)
```

Comment:
This won't work if it's a Tensor though, right?

Reply:
Yes, this won't work. Will there be a case when simulator returns a tensor object? In my experiments it has always been a numpy array. We need to typecast to int32 as tensor won't support uint32

Reply:
When it's running in GPU2GPU mode.